subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java

```diff
@@ -145,10 +145,11 @@ protected void onApply(final Project project) {
     }
 
     private void configureIdeaWorkspace(final Project project) {
+        final IdeaWorkspace workspace = project.getObjects().newInstance(IdeaWorkspace.class);
+        ideaModel.setWorkspace(workspace);
+
         if (isRoot()) {
-            final IdeaWorkspace workspace = project.getObjects().newInstance(IdeaWorkspace.class);
             workspace.setIws(new XmlFileContentMerger(new XmlTransformer()));
-            ideaModel.setWorkspace(workspace);
```

Comment:
Is it a meaningful difference that we now also do this when `isRoot()` returns `false`?

Reply:
The previous behavior was to:
- Always create a IdeaWorkspace with `new`
- In the root build, instantiate a new IdeaWorkspace with ObjectFactory
- In the root build, set iws
- In the root build, overwrite the workspace created with `new`

The new behavior is:
- In all builds, instantiate a new IdeaWorkspace with ObjectFactory
- In the root build, set iws

It seems to me that the IdeaWorkspace isn't valid for non-root projects, but it's always been set to something. A more correct solution might be to make it null or only add this extension to the root project. I only changed this around to avoid the use of `new`.